syntax = "proto3";

package openai_prompt_service;

message Message {
  enum Role {
    SYSTEM = 0;
    ASSISTANT = 1;
    USER = 2;
  }

  Role role = 1;
  string content = 2;
}

message OpenAIPromptRequest {
  repeated Message messages = 1;
  string model = 2;
  int32 max_tokens = 3;
  float temperature = 4;
  string api_token = 5;
}

message ResponseData {
  string message = 1;
  string error = 2;
}

message OpenAIPromptResponse {


  bool success = 1;
  ResponseData data = 2;
  float execution_time = 3;
}


service OpenAIPromptService {
  rpc CompletePrompt (OpenAIPromptRequest) returns (OpenAIPromptResponse) {}
}
